internal/provider/factory_utils.go

```diff
@@ -228,13 +228,31 @@ func (f RetryableClientFactory) CreateRetryableClient() *http.Client {
 		retryClient.RetryMax = *f.maxRetries
 	}
 
+	retryClient.ErrorHandler = customErrorHandler
+
 	// Create a logger for retryablehttp
 	// This logger will be used to send retryablehttp's internal logs to tflog
 	retryClient.Logger = logger
 
 	return retryClient.StandardClient()
 }
 
+func customErrorHandler(resp *http.Response, err error, retries int) (*http.Response, error) {
+	if resp != nil {
+		body := resp.Body
+		defer body.Close()
+		if resp.StatusCode == 429 {
```

Comment:
Can we add the same thing for 5xx errors? I suspect our existing 5xx error messages aren’t very descriptive, and it’s not clear whether a failure was due to a 429 or a 5xx.